### PR TITLE
/subscribe endpoint

### PR DESCRIPTION
**Pull request by tomrf1**
Adds a new endpoint for one-click subscribes from email links.
See [IDAPI PR](https://github.com/guardian/identity/pull/2397) for details of how this works.

I've refactored the existing /unsubscribe functionality, because both subscribing and unsubscribing work the same way.
- `src/server/routes/unsubscribe.ts` is now `src/server/routes/subscriptions.ts`, and handles both endpoints
- `UnsubscribeSuccessPage` is now `SubscriptionSuccessPage`
- `UnsubscribeErrorPage` is now `SubscriptionErrorPage`

And we now use `type SubscriptionAction = 'unsubscribe' | 'subscribe'` to vary behaviour across the handlers/components.

Tested in CODE:
Screenshot 2023-05-15 at 15 22 59

### CODE REVIEW

File: src/client/pages/SubscriptionErrorPage.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (tomrf1) [inline src/client/pages/SubscriptionErrorPage.tsx:1]: This file was called `UnsubscribeErrorPage.tsx`

File: src/client/pages/SubscriptionError.stories.tsx [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
-  title: 'Pages/UnsubscribeSuccess',
-  component: UnsubscribeSuccess,
+  title: 'Pages/SubscriptionError',
+  component: SubscriptionError,
   parameters: { layout: 'fullscreen' },
 } as Meta;
 
 export const Default = () => (
-  <UnsubscribeSuccess accountManagementUrl="#" returnUrl="#" />
+  <SubscriptionError accountManagementUrl="#" action={'unsubscribe'} />
```
Comment (coldlink) [inline src/client/pages/SubscriptionError.stories.tsx:14]: worth adding an additional story showing SubscriptionError with `action={'subscribe'}`

File: src/client/pages/SubscriptionSuccess.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export const Default = () => (
+  <SubscriptionSuccess
+    accountManagementUrl="#"
+    returnUrl="#"
+    action={'unsubscribe'}
+  />
+);
+Default.story = {
+  name: 'with defaults',
+};
```
Comment (coldlink) [inline src/client/pages/SubscriptionSuccess.stories.tsx:22]: and another additional story here for `action={'subscribe'}`

**Review by coldlink — COMMENTED** [review]
just a couple of comments!

**Pull request merged**